DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -200,6 +200,58 @@ private async ValueTask<TokenKind> GetTokenKindAsync(CancellationToken cancellat
         return Guild.Parse(response);
     }
 
+    public async IAsyncEnumerable<Data.Thread> GetGuildChannelThreadsAsync(
+        string channelId,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        int currentOffset = 0;
+
+        var completeThreads = new List<JsonElement>();
```

Comment:
Since this method returns `IAsyncEnumerable<...>`, you just yield elements asynchronously, instead of buffering them in a list (see further comments)

Reply:
Resolved in 80cfb15.